### PR TITLE
surgery update (animal and moth surgery)

### LINKED ISSUE #874: some species organs dont register
## Description
In surgery some species organs dont register for removal, one example is slime, which cannot have organs removed, including brain
**Reproduction**
surgery on slime, try to remove brain / organs

### LINKED ISSUE #878: Giving a human vox lungs doesnt make them need nitrogen
## Description
title

**Reproduction**
rip out vox lungs
put in human
human breath oxygen still

### PR DESCRIPTION
**Pull request by deltanedas**
## About the PR
- a lot of animals can be operated on now (mice, mothroach/maggot???, monkeys, kobolds probably)
- fixed a bunch of organs not being transplantable like diona brain, arachnid heart, all stomachs, some others
- YOU CAN NOW TURN PEOPLE INTO MOTHS BY GIVING THEM MOTH STOMACHS
- YOU CAN NOW TURN PEOPLE INTO VOXES BY GIVING THEM VOX LUNGS

make the ultimate being that has to breathe nitrogen and eat cloth

## Why / Balance
"Sec to surgery", the clown blurted out.
"He's turning her into a mothroach!"

more manmade abominations like mice that eat clothes 
vox lung bug is fixed

fixes #878
fixes #874

## Technical details
added new thing in body system to check if an organ can be attached (without this check, you would be able to do things like attach a head to mice. too based for this pr)

## Media
idk

vox lungs in a human works!!!

## Requirements
- [X] I have read and am following the [Pull Request and Changelog Guidelines](https://docs.spacestation14.com/en/general-development/codebase-info/pull-request-guidelines.html).
- [X] I have added media to this PR or it does not require an ingame showcase.

## Breaking changes
:trollface:

**Changelog**
:cl:
- add: You can now operate on a bunch of animals like mice, spiders and monkeys.
- add: You can now remove and install stomachs, including special diets like moths or lizards!
- fix: Fixed not being able to reinstall some species organs.
- fix: Fixed harpies being immune to the scalpel.

## Summary by CodeRabbit

- **New Features**
	- Enhanced body part management with new event subscriptions for initialization and removal.
	- Added new surgical actions and steps for attaching and removing body parts.
	- Introduced new organ components for various species, improving gameplay mechanics.
	- New Harpy interaction capabilities added with UI components for surgery.
	- New entities and components for animal and mob interactions, enhancing gameplay complexity.

- **Bug Fixes**
	- Improved validation logic for surgery conditions, ensuring correct attachment checks.

- **Documentation**
	- Updated documentation for new fields and components added across various entities.

### CODE REVIEW

File: Resources/Prototypes/Body/Organs/vox.yml [hunk 1]
```diff
@@ -2,6 +2,7 @@
   id: OrganVoxLungs
   parent: OrganHumanLungs
   suffix: "vox"
+  name: vox longs # GoobStation
```
[the remark this answers is not in the document]
Comment (deltanedas) [reply Resources/Prototypes/Body/Organs/vox.yml]: wtf real

**Comment by deltanedas** [comment]
whoever changed the ai slop bot to use the furry prompt its actually good now highly real

**Comment by gluesniffler** [comment]
killer beat me to it lmao, ill be porting your changes to the EE branch

File: Content.Shared/Body/Systems/SharedBodySystem.Parts.cs [hunk 2]
```diff
@@ -565,6 +565,18 @@ public bool CanAttachPart(
...
+    /// </summary>
+    public bool CanAttachToSlot(
+        EntityUid parentId,
+        string slotId,
+        BodyPartComponent? parentPart = null)
+    {
+        return Resolve(parentId, ref parentPart, logMissing: false)
+            && parentPart.Children.ContainsKey(slotId);
+    }
+
```
Comment (gluesniffler) [inline Content.Shared/Body/Systems/SharedBodySystem.Parts.cs:579]: Whats the difference between using these and CanAttachpart?

File: Resources/Prototypes/Body/Prototypes/primate.yml [hunk 3]
```diff
@@ -3,11 +3,19 @@
   name: "primate"
   root: torso
   slots:
+    head: # GoobStation: put pun pun into a humans body
+      part: HeadAnimal
```
Comment (gluesniffler) [inline Resources/Prototypes/Body/Prototypes/primate.yml:7]: Holy based.

**Comment by yglop** [comment]
Surgery is just a week away ancestors were saying. And here we are. He did it.
He did it (1)

File: Resources/Prototypes/Entities/Mobs/NPCs/slimes.yml [hunk 4]
```diff
@@ -92,6 +92,11 @@
   - type: Body
     prototype: Slimes
     requiredLegs: 1
+  - type: SurgeryTarget # GoobStation
+  - type: UserInterface # GoobStation
+    interfaces:
+      enum.SurgeryUIKey.key:
+        type: SurgeryBui
```
Comment (gluesniffler) [inline Resources/Prototypes/Entities/Mobs/NPCs/slimes.yml:99]: This seems to be the source of most of the test fails. Not sure whats up with slime mobs but apparently they do not like the UserInterface component.

File: Content.Shared/Body/Systems/SharedBodySystem.Parts.cs (hunk 2 above)
Comment (deltanedas) [reply Content.Shared/Body/Systems/SharedBodySystem.Parts.cs:579]: theres no partId to use at the stage this is called from

**Comment by deltanedas** [comment]
durk instamerge NOW!

File: Resources/Prototypes/Entities/Mobs/NPCs/slimes.yml (hunk 4 above)
Comment (deltanedas) [reply Resources/Prototypes/Entities/Mobs/NPCs/slimes.yml]: typographical error

**Comment by deltanedas** [comment]
forgot to mention but of course this PRs patch can be used under CC-0 do whatever you want on other forks

**Pull request merged**